frontend/src/components/ProjectCardGrid/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect } from 'react'
+import { useInView } from 'react-intersection-observer'
+
+import { meiliIndex } from '@utils/meili'
+import { Filter } from 'meilisearch'
+import ProjectCard, { IntersectionObserverRef } from '../ProjectCard'
+import styles from './index.module.scss'
+
+export const cardsPerRow = 3
+const LIMIT = cardsPerRow * 6
+
+export interface Project {
+  bom: object
+  gitHash: string
+  id: string
+  ownerName: string
+  projectName: string
+  readme: string
+  repoId: string
+  repoName: string
+  summary: string
+}
+
+interface CardsGridProps {
+  projects: Project[]
+  intersectionObserverRef?: IntersectionObserverRef
+}
+
+const CardsGrid = ({ projects, intersectionObserverRef }: CardsGridProps) => {
+  return (
+    <div className={styles.cardsGrid} data-cy="cards-grid">
+      {projects?.map((project, index) => (
+        <ProjectCard
+          {...project}
+          key={project.id}
+          intersectionObserverRef={
+            index === projects.length - cardsPerRow * 2
+              ? intersectionObserverRef
+              : null
+          }
+        />
+      ))}
+    </div>
+  )
+}
+
+export const getKey =
+  ({ query, filter }: { query: string; filter?: Filter }) =>
+  (pageIndex: number, previousPageData: Project[]) => {
+    if (previousPageData && !previousPageData.length) {
+      // reached the end
+      return null
+    }
+    return { query, offset: pageIndex * LIMIT, limit: LIMIT, filter }
+  }
+
+export interface GridFetcherArgs {
+  filter?: Filter
+  limit?: number
+  offset?: number
+  query?: string
+}
+
+export const gridFetcher = async ({
+  filter,
+  offset = 0,
+  query = '*',
+  limit = LIMIT,
+}: GridFetcherArgs) => {
+  const searchResult = await meiliIndex.search(query, { limit, offset, filter })
+  return searchResult.hits as Project[]
+}
+
+/**
+ * Update useSWRInfinite size when the user reaches two rows from the end of the grid.
+ * @param  setSize - the function returned by useSWRInfinite.
+ */
+export const useUpdateBeforeReachingLimit = (
+  setSize: (size: (_size: number) => number) => Promise<Project[][] | undefined>,
```

Comment:
Please separate the types out to make this more readable.

Reply:
Here's an example: https://github.com/kitspace/kitspace-v2/blob/8fdada80cfb49920ebcef379a587817fc1b37f44/frontend/src/utils/index.ts#L98-L115